core/templates/dev/head/services/assets-backend-api.service.ts

```diff
@@ -188,19 +195,14 @@ angular.module('oppia').factory('AssetsBackendApiService', [
           contentType: false,
           type: 'POST',
           dataType: 'text',
-          dataFilter: function(data) {
-            // Remove the XSSI prefix.
-            var transformedData = data.substring(5);
-            return JSON.parse(transformedData);
-          },
+          dataFilter: removeXSSIPrefix,
         }).done(function(response) {
           if (successCallback) {
             successCallback(response);
           }
         }).fail(function(data) {
           // Remove the XSSI prefix.
-          var transformedData = data.responseText.substring(5);
-          var parsedResponse = angular.fromJson(transformedData);
+          var parsedResponse = removeXSSIPrefix(data.responseText);
```

Comment:
Thanks for creating a separate for this one! :)

Reply:
Don't mention it! :smile: